### PR TITLE
Add custom reports modal

### PR DESCRIPTION
**Pull request by kshepard**
custom-reports-1
custom-reports-2
custom-reports-3
custom-reports-4

### CODE REVIEW

File: web/app/scripts/custom-reports/custom-reports-modal-controller.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+            {
+                label: 'Month of Year',
+                value: 'month',
+                type: 'Time'
+            },
+            {
+                label: 'Week of Year',
+                value: 'week',
+                type: 'Time'
+            }
```
Comment (ddohler) [inline web/app/scripts/custom-reports/custom-reports-modal-controller.js]: The endpoint also supports years, not sure if we actually want to let users select that though.

File: web/app/scripts/custom-reports/custom-reports-modal-controller.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                    params = _.extend(params, crosstabsParams);
+
+                    // Ensure the page limit parameter is not set
+                    if (params.limit) {
+                        delete params.limit;
+                    }
+
+                    // The crosstabs endpoint uses `record_type_id` rather than `record_type`.
+                    // Renaming it here, though the name should probably be standardized.
+                    params.record_type_id = params.record_type;
```
Comment (ddohler) [inline web/app/scripts/custom-reports/custom-reports-modal-controller.js]: Yeah, standardizing on `record_type` is probably better; I was thinking in Django terms but from the point of view of the Javascript it should just be `record_type`, not `record_type_id`.

File: web/app/scripts/custom-reports/custom-reports-modal-controller.js (hunk 1 above)
Comment (kshepard) [inline web/app/scripts/custom-reports/custom-reports-modal-controller.js]: Oh, thanks, I missed that one. I think it's worth adding. If we see some performance issues while testing, we can always strip it out again.

File: web/app/scripts/custom-reports/custom-reports-modal-controller.js (hunk 2 above)
Comment (kshepard) [inline web/app/scripts/custom-reports/custom-reports-modal-controller.js]: Cool, I kind of figured, but wasn't sure if there was a reason I was missing. I'll remove this and update the backend code.

**Comment by ddohler** [comment]
+1

File: web/app/scripts/custom-reports/custom-reports-modal-controller.js (hunk 2 above)
Comment (kshepard) [inline web/app/scripts/custom-reports/custom-reports-modal-controller.js]: Oh, I think it's just the docstring that was saying `record_type_id` on the backend -- the param actually being used is `record_type`. Easy fix!

Comment (ddohler) [inline web/app/scripts/custom-reports/custom-reports-modal-controller.js]: Hah, cool, sorry for the confusion!

**Pull request merged**